controller/weak_supervision/weak_supervision_service.py

```diff
@@ -34,7 +36,10 @@ def calculate_quality_after_labeling(project_id: str, labeling_task_id: str, use
     return service_requests.post_call_or_raise(url, data)
 
 
-def calculate_stats_after_source_run(project_id: str, source_id: str, user_id: str) -> Any:
+@debounce(2)
```

Comment:
if i'm not mistaken this was intentionally not debounced to have an "on demand" execution -> debounce will start a thread and not wait for the return. Is this wanted?

Reply:
The data is sent from the SDK in smaller batches, so I don't want to recompute the stats every time (this would again result in an overload). 2 seconds for debounce should be ok I guess. An alternative would be to add another function calling this, which receives the debounce, and only having the callback execute this very function.

Reply:
nvm I changed it so that only the callbacks have a debounce